rainier-core/src/main/scala/com/stripe/rainier/core/Categorical.scala

```diff
@@ -99,10 +99,27 @@ final case class Binomial(p: Real, k: Int) extends Distribution[Int] {
   val multi: Multinomial[Boolean] =
     Categorical.boolean(p).toMultinomial(k)
 
-  def generator: Generator[Int] =
-    multi.generator.map { m =>
+  def generator: Generator[Int] = {
+    val poissonGenerator = Poisson(p * k).generator.map { _.min(k) }
+    val normalGenerator = Normal(k * p, k * p * (1 - p)).generator.map {
+      _.toInt.max(0).min(k)
+    }
+    val binomialGenerator = multi.generator.map { m =>
       m.getOrElse(true, 0)
     }
+    Generator.from {
+      case (r, n) =>
+        if (k >= 100 && k * n.toDouble(p) <= 10) { poissonGenerator.get(r, n) } else if (k >= 100 && k * n
```

Comment:
Seems like you'd do well to stash `n.toDouble(p)` somewhere instead of recomputing it a bunch of times

Reply:
👍 done